libs/math/include/math/TQuatHelpers.h

```diff
@@ -259,10 +260,11 @@ class TQuatFunctions {
             return normalize(lerp(d < 0 ? -p : p, q, t));
         }
         const T npq = std::sqrt(dot(p, p) * dot(q, q));  // ||p|| * ||q||
-        const T a = std::acos(math::clamp(absd / npq, T(-1), T(1)));
+        const T cos_a = math::clamp(absd / npq, T(-1), T(1));
+        const T a = std::acos(cos_a);
         const T a0 = a * (1 - t);
         const T a1 = a * t;
-        const T sina = sin(a);
+        const T sina = std::sqrt(T(1) - cos_a * cos_a);
```

Comment:
I don't think you get at all the same precision. `1-x` can be a precision killer if `x` is very small. And then `sqrt`.

Reply:
1-x is actually quite accurate in a relative sense when x is small, because you cannot have catastrophic cancellation. Instead, the result saturates to 1. In this case, the accuracy in the near worst-case about the same, compared to the original formula using double precision:

https://godbolt.org/z/GPfb465Wj

FWIW: These types of identities are commonly used to compute rotation matrices in linear algebra eigensolvers due to their stability and speed.

Reply:
sqrt is very well behaved numerically, and most platforms that I can think of actually implement it correctly rounded for IEEE types.

Reply:
Okay I don't have enough expertise here; but I buy it.